### ISSUE TITLE
[BUG] tracing.stopChunk: Tracing is already stopping

### ISSUE THREAD

**Issue by pbiggar** [microsoft/playwright#10190]
**Context:**
## System:
 - OS: macOS 12.0.1
 - Memory: 16.38 GB / 64.00 GB
## Binaries:
 - Node: 16.11.1 - /usr/local/bin/node
 - npm: 8.0.0 - /usr/local/bin/npm
## Languages:
 - Bash: 5.1.8 - /usr/local/bin/bash

**Describe the bug**

When I set playwright to retry and trace, tracing fails with the following error:

```
tracing.stopChunk: Tracing is already stopping

    at /home/dark/app/integration-tests/node_modules/@playwright/test/lib/index.js:433:48
    at Object._baseTest.extend._setupContextOptionsAndArtifacts.auto [as fn] (/home/dark/app/integration-tests/node_modules/@playwright/test/lib/index.js:432:40)
    at Fixture.teardown (/home/dark/app/integration-tests/node_modules/@playwright/test/lib/fixtures.js:101:7)
    at Fixture.teardown (/home/dark/app/integration-tests/node_modules/@playwright/test/lib/fixtures.js:92:40)
    at FixtureRunner.teardownScope (/home/dark/app/integration-tests/node_modules/@playwright/test/lib/fixtures.js:272:11)
    at /home/dark/app/integration-tests/node_modules/@playwright/test/lib/workerRunner.js:98:7
```

<details>
This came from this CI run: https://app.circleci.com/pipelines/github/darklang/dark/1755/workflows/9969561f-7202-41f3-80bf-205dc7f5677b/jobs/17850/parallel-runs/0?filterBy=FAILED (this is public, it should be visible by logging in using any github account).
</details>

Config file:
```
 $ cat integration-tests/playwright.config.ts
// Config file for integration tests
import { PlaywrightTestConfig, devices } from "@playwright/test";

const config: PlaywrightTestConfig = {
  testDir: ".",
  testMatch: "tests.ts",
  expect: {
    // timeout: 5000,
  },
  timeout: 15000,
  use: {
    // actionTimeout: 1000,
    headless: true,
    // Ideally this would be retain-on-failure, but it fails sometimes
    trace: "on",
    screenshot: "off",
    video: "on",
  },
  reporter: [
    ["list"],
    ["json", { outputFile: "rundir/test_results/integration_tests.json" }],
    ["junit", { outputFile: "rundir/test_results/integration_tests.xml" }],
  ],
};

export default config;
```

I'm not sure if the test file is relevant, but if so it's available here: https://github.com/darklang/dark/blob/45512731bbce7df6a5fd4d12ff7f781e4727fa6f/integration-tests/tests.ts

**Comment by aslushnikov** [comment]
Looking at the code, there's a race between our instrumentation of `onWillCloseContext` and cleaning up leftover contexts.

**Comment by aslushnikov** [comment]
@pbiggar oh actually, this has been fixed recently in 3c1aaa5338f9822d39a4c69c4ddd19acc12d0530

So will be working fine in upcoming 1.17
